pkg/rtc/rtc.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package rtc
+
+import (
+	"errors"
+	"os"
+	"time"
+
+	"golang.org/x/sys/unix"
+)
+
+type RTC struct {
+	file *os.File
+}
+
+func OpenRTC() (*RTC, error) {
+	devs := []string{
+		"/dev/rtc",
+		"/dev/rtc0",
+		"/dev/misc/rtc0",
+	}
+
+	for _, dev := range devs {
+		f, err := os.Open(dev)
+		if err == nil || !os.IsNotExist(err) {
```

Comment:
I'm not so sure about this. I am pretty sure on any error, f will be nil. You're going to return a nil pointer in the RTC I bet.

Reply:
The idea here was to return early if either opening the device node succeeded (`err == nil`). Or if there is a permission error or any other non-recoverable error for which it wouldn't make sense trying to open the other device nodes. Maybe using `os.IsNotExist` here doesn't convey that idea correctly?

As for returning a f as `nil` this is also what `os.Open` does in case of an error. I'd expect any user of `OpenRTC` to check `err` before accessing `f` - like they would for the return values of `os.Open`. Or should we rather return an empty `rtc.RTC{}`?

Reply:
Sorry, I misunderstood your comment. Currently `OpenRTC` is not returning `nil` in this case but an `RTC` with `RTC.file == nil`. This will surely lead to problems. I'll change it to return `nil`. Thanks.

Reply:
yeah I confirmed it. 
Try this
touch /tmp/noaccess
chmod 0 /tmp/noaccess
run this program:
package main

import "fmt"
import "os"

func main() {
	f, err := os.Open("/tmp/noaccess")
	fmt.Printf("%v %v", f, err)
}

you get
<nil> open /tmp/noaccess: permission denied

so just test for err != nil here

Reply:
oh that damned comment box removed the result :-(
"<nil> open /tmp/noaccess: permission denied"

Reply:
I'm really hating this fancy format shit.
`<nil> open /tmp/noaccess: permission denied`

Reply:
But if we're testing for `err != nil` we would never end up trying  the other possible device nodes in `devs`, no?

I now changed it to

```go
                if err == nil {
                        return &RTC{f}, err
                } else if !os.IsNotExist(err) {
                        return nil, err
                }
```

In this case we would be retrying with the other `/dev/...` file in case the first doesn't exist.

Reply:
Or maybe the intention should be made more explicit? i.e.

```go
                if os.IsNotExist(err) {
                        continue // try next RTC device node
                } else if err != nil {
                        return nil, err
                } else {
                        return &RTC{f}, err
                }
```